railties/test/application/loading_test.rb

```diff
@@ -279,12 +279,12 @@ class User
     require "#{rails_root}/config/environment"
 
     get "/c"
-    assert_equal "3", last_response.body
+    assert_equal "5", last_response.body
```

Comment:
In the routes file in this test:
```
$counter ||= 1
$counter  *= 2
Rails.application.routes.draw do
  get '/c', to: lambda { |env| User.name; [200, {"Content-Type" => "text/plain"}, [$counter.to_s]] }
end
```
This doesn't add 2 and 4 more reloads. I believe the change is caused by `routes_reloader.execute` which reloads all routes by re-evaluating route files. This also only happens in development mode.